src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -7,57 +7,57 @@
 using Microsoft.CodeAnalysis.CodeCleanup;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Extensions;
-using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Commanding;
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
 using Roslyn.Utilities;
-using static Microsoft.CodeAnalysis.Options.OptionServiceFactory;
 using VSCommanding = Microsoft.VisualStudio.Commanding;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Formatting
 {
     internal partial class FormatCommandHandler : ForegroundThreadAffinitizedObject
     {
-        private bool _infoBarOpen = false;
         public VSCommanding.CommandState GetCommandState(FormatDocumentCommandArgs args)
         {
             return GetCommandState(args.SubjectBuffer);
         }
 
-        private void ShowGoldBarForCodeCleanupConfiguration(Document document)
+        private void ShowGoldBarForCodeCleanupConfigurationIfNeeded(Document document)
         {
             AssertIsForeground();
 
-            // If the gold bar is already open, do not show
-            if (_infoBarOpen)
+            var workspace = document.Project.Solution.Workspace;
+
+            // if info bar was shown before, no need to show it again
+            if (workspace.Options.GetOption(CodeCleanupOptions.NeverShowCodeCleanupInfoBarAgain, document.Project.Language) ||
+                workspace.Options.GetOption(CodeCleanupOptions.CodeCleanupInfoBarShown, document.Project.Language))
```

Comment:
:memo: Also no need to show the gold bar if code cleanup is enabled (`CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting`). Technically this condition is handled by `CodeCleanupOrFormat` but it wouldn't hurt IMO to be explicit here.

Reply:
Not true , gold bar need to shown even if enabled because they could configure for more changes

Reply:
hmm. there seems some confusion. if the option is on, infobar will be never shown. this code path doesn't get hit if option is on.

Reply:
nevermind , i synced with Sam ...yes if the user went in and enabled the features we don't show the infobar again for them